backend/accounts/baseaccount.go

```diff
@@ -168,14 +213,33 @@ func (account *BaseAccount) GetAndClearProposedTxNote() string {
 
 // SetTxNote implements accounts.Account.
 func (account *BaseAccount) SetTxNote(txID string, note string) error {
-	if err := account.notes.SetTxNote(txID, note); err != nil {
-		return err
+	for idx, notes := range account.notes {
+		// Store the note, delete it from legacy notes.
+		storeNote := note
+		if idx > 0 {
+			storeNote = ""
+		}
+		if err := notes.SetTxNote(txID, storeNote); err != nil {
+			return err
```

Comment:
I would ignore and continue here for idx > 0. Maybe a more straightforward way - tbh I didn't get `storeNote = ""` at the first second - is to split new notes vs legacy for clarity:

```go
// The notes slice is guaranteed to have at least one element by BaseAccount.Initialize. 
if err := account.notes[0].SetTxNote(txID, note); err != nil {
  return err
}
// Delete the notes in legacy files. Don't really care if it fails.
for i, notes := range account.notes[1:] {
  if err := notes.SetTxNote(txID, ""); err != nil {
    log.Printf("Can't delete a note from a legacy file idx=%d: %v", i, err)
  }
}
return nil
```

Reply:
done